src/badguy/willowisp.cpp

```diff
@@ -91,9 +94,55 @@ WillOWisp::WillOWisp(const ReaderMapping& reader) :
 void
 WillOWisp::finish_construction()
 {
-  if (get_walker() && get_walker()->is_running()) {
-    m_mystate = STATE_PATHMOVING_TRACK;
+  if (!get_path())
+    init_path_pos(m_col.m_bbox.p1());
+
+  if (has_valid_path())
+  {
+    const auto& nodes = get_path()->get_nodes();
+    if (!nodes.empty())
+    {
+      if (m_starting_node < 0)
+      {
+        m_starting_node = 0;
+      }
+      if (m_starting_node >= static_cast<int>(nodes.size()))
```

Comment:
It's not mandatory, but I feel like moving 

`static_cast<int>(nodes.size())` into a variable and using that could be beneficial here.